src/main/java/org/elasticsearch/index/mapper/core/TypeParsers.java

```diff
@@ -270,13 +270,11 @@ public static void parseField(AbstractFieldMapper.Builder builder, String name,
             } else if (propName.equals("include_in_all")) {
                 builder.includeInAll(nodeBooleanValue(propNode));
                 iterator.remove();
-            } else if (propName.equals("postings_format")) {
-                String postingFormatName = propNode.toString();
-                builder.postingsFormat(parserContext.postingFormatService().get(postingFormatName));
+            } else if (propName.equals("postings_format") && parserContext.indexVersionCreated().before(Version.V_2_0_0)) {
+                // ignore in 1.x reject in 2.x
```

Comment:
This comment is a bit misleading.  I think it should be "ignore for pre 2.0 indexes"? The way it is written I thought this implied this change would be backported.